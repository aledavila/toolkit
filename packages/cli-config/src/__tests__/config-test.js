/**
 * @jest-environment node
 */

'use strict';

const path = require('path');

const fixtures = path.resolve(__dirname, '../__fixtures__');

describe('config', () => {
  let Config;
  let name;

  beforeEach(() => {
    Config = require('../config');
    name = 'toolkit';
  });

  describe('load', () => {
    describe('with no config', () => {
      it('should return an empty config', () => {
        const fixture = path.join(fixtures, 'with-no-config');
        const { noConfig } = Config.load({ name, cwd: fixture });
        expect(noConfig).toBeDefined();
        expect(noConfig).toBe(true);
      });
    });

    describe('with invalid config', () => {
      it('should report a validation error', () => {
        const fixture = path.join(fixtures, 'with-invalid-config');
        const { error } = Config.load({ name, cwd: fixture });
        expect(error).toBeDefined();
      });
    });

    describe('with config', () => {
      it('should return the config and plugins to load', () => {
        const fixture = path.join(fixtures, 'with-config');
        const { config, filepath, plugins } = Config.load({
          name,
          cwd: fixture,
        });
        expect(filepath).toBeDefined();
        expect(config).toMatchSnapshot();
        expect(plugins).toMatchSnapshot();
      });
    });
  });
});
