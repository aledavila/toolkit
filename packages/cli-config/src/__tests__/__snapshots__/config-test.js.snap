// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config load with config should return the config and plugins to load 1`] = `
Object {
  "all": [Function],
  "has": [Function],
  "why": [Function],
}
`;

exports[`config load with config should return the config and plugins to load 2`] = `
Array [
  Object {
    "name": "../modules/a.js",
    "options": Object {},
    "owner": Object {
      "root": true,
    },
    "plugin": [Function],
    "type": "plugin",
  },
]
`;
